sdk/include/opentelemetry/sdk/metrics/meter.h

```diff
@@ -20,254 +25,360 @@ class Meter final : public metrics_api::Meter, public std::enable_shared_from_th
     library_version_ = library_version;
   }
 
+  /**
+   * Creates a Counter with the passed characteristics and returns a shared_ptr to that Counter.
+   *
+   * @param name the name of the new Counter.
+   * @param description a brief description of what the Counter is used for.
+   * @param unit the unit of metric values following https://unitsofmeasure.org/ucum.html.
+   * @param enabled a boolean value that turns on or off the metric instrument.
+   * @return a shared pointer to the created Counter.
+   * @throws IllegalArgumentException if {@code name} is null
```

Comment:
Is this copied from Java and is it valid here?

Reply:
Good catch! Yes, it is. I updated the comments to better reflect the actual code.